src/adapter/src/coord/cluster_scheduling.rs

```diff
@@ -102,12 +105,22 @@ impl Coordinator {
             );
             let decisions = min_refresh_mv_write_frontiers
                 .into_iter()
-                .map(|(cluster_id, min_refresh_mv_write_frontier)| {
-                    (
-                        cluster_id,
-                        min_refresh_mv_write_frontier.less_than(&local_read_ts),
-                    )
-                })
+                .map(
+                    |(cluster_id, rehydration_time_estimate, min_refresh_mv_write_frontier)| {
+                        (
+                            cluster_id,
+                            // We are just checking that
+                            // write_frontier < local_read_ts + rehydration_time_estimate
+                            min_refresh_mv_write_frontier.less_than(
+                                &(local_read_ts.step_forward_by(
+                                    &rehydration_time_estimate.try_into().expect(
+                                        "`rehydration_time_estimate` is checked during planning",
+                                    ),
+                                )),
+                            ),
+                        )
+                    },
```

Comment:
nit: I find this a bit easier to read if you break things out into local variables and avoid right-ward drift, e.g.
```
let rehydration_estimate = rehydration_time_estimate.try_into().expect("checked during planning");
let time_to_schedule = local_read_ts.step_forward_by(rehydration_estimate);
let should_schedule = min_refresh_mv_write_frontier.less_than(time_to_schedule);

(cluster_id, should_schedule)
```

This is subjective though so totally up to you!

Reply:
Yeah, changed it, looks better this way.